---
title: "FMO3_Genotype"
author: "Kristen James"
date: "2022-09-29"
output: html_document
---

# About
Add FMO3 genotype data to analysis. Ask if TMAO response differs by FMO3 genotype. 

See "FMO3_SNP_forExternalShare_210407.rmd" for further info copied below: 
We are interested in assessing whether genotype at 2 FMO3 missence SNPs, rs2266780 and rs2266782, is associated with TMAO levels. We hypothesize that the SNPs will alter the activity of the FMO3 enzyme and affect TMAO levels. 

Because we have 2 genotyped SNPs and because these SNPs are often inherited together we can assess our question in multiple ways, including: 

- Is there an association between rs2266780 and TMAO levels?
- Is there an association between rs2266782 and TMAO levels?
- Is there an association between combined genotype and TMAO levels?

The SNP data contains the following columns and is described.

**Data Dictionary** for SNP list:

- **MinorMajorCode_2** - rs2266782, 1=major allele, 2=het, 3=minor allele, G>A where G is major
- **MinorMajorCode_0** - rs2266780, 1=major allele, 2=het, 3=minor allele, A>G where A is major
- **rs2** - same as MinorMajorCode_2
- **rs0** - same as MinorMajorCode_0
- **FMO3geno** - 1st letter is SNP rs2266782, 2nd letter is SNP rs2266780; A = homo major, H = hetero, B = homo SNP; haplotype
- **FMO3_nRiskAlleles** - sum of the number of SNP alleles between both SNPs (0-4)
- **rs0_WT_V** - 1=WT, 2=either 1 or 2 SNP alleles
- **rs2_WT_V** - 1=WT, 2=either 1 or 2 SNP alleles

**Genotyping Info**:

The SNPs in the genotyping file are coded such that, 1=major allele, 2=het, 3=minor allele.

- For rs2266782 G>A where G is major
- For rs2266780 A>G where A is major

```{r setup, include=FALSE}
# Libraries
library(ggplot2)
library(ggpubr)
library(tableone)
library(HardyWeinberg) #HWChisq
library(car)

# Clear environment to start fresh (if necessary)
rm(list = ls())

# Data from TMAO_Meal_Response_Focused.rmd
#dfwide <- readRDS("../../Data/Processed/dfwide_metavariables_n97_inflam.rds")
dfwide <- read.csv("../../Data/Processed/dfwide_metavariables_n97_inflam_eth.csv", row.names = 1)

# SNP data 
SNP <- read.csv("../../Data/from_FL100V2/FL100_FMO3SNP_Results_Final_8_15_19.csv", row.names = 1)

# Ethnicity data
eth <- read.csv("../../Data/Raw/demographic/ethnicity_210715.csv",row.names = 1)

```

Using the SNP genotypes, create categorical variables. 

```{r SNP categories}
#
# SNP modifications
#
# Factor SNPs
table(SNP$MinorMajorCode_2)
table(SNP$MinorMajorCode_0)
str(SNP$MinorMajorCode_2) # currently numeric
SNP$rs2 <- factor(SNP$MinorMajorCode_2, levels = c(1,2,3))
SNP$rs0 <- factor(SNP$MinorMajorCode_0, levels = c(1,2,3))
str(SNP$rs2)

# Make LD SNP combined variable
# There are 9 possibilities
# 1=A homo wildtype; 2=H heterozygous; 3=B homo variant
# rs2266782 A H B
# rs2266780 A H B
# AA AH AB HA HH HB BA BH BB
# 1  2  3  4  5  6  7  8  9
SNP$FMO3geno <- ifelse(SNP$rs2 == 1 & SNP$rs0 == 1, 1,
                       ifelse(SNP$rs2 == 1 & SNP$rs0 == 2, 2,
                              ifelse(SNP$rs2 == 1 & SNP$rs0 == 3, 3,
                                     ifelse(SNP$rs2 == 2 & SNP$rs0 == 1, 4,
                                            ifelse(SNP$rs2 == 2 & SNP$rs0 == 2, 5,
                                                   ifelse(SNP$rs2 == 2 & SNP$rs0 == 3, 6,
                                                          ifelse(SNP$rs2 == 3 & SNP$rs0 == 1, 7,
                                                                 ifelse(SNP$rs2 == 3 & SNP$rs0 == 2, 8,
                                                                        ifelse(SNP$rs2 == 3 & SNP$rs0 == 3, 9, 10)
                                                                        )
                                                                 )
                                                          )
                                                   )
                                            )
                                     )
                              )
                       )

# Confirm worked as expected and factor
table(SNP$FMO3geno)
SNP$FMO3geno <- as.factor(SNP$FMO3geno)
str(SNP$FMO3geno)
# Revalue factors; allele order is rs2, rs0; A=WT,H=het,B=variant
levels(SNP$FMO3geno)[levels(SNP$FMO3geno)=="1"] <- "AA"
levels(SNP$FMO3geno)[levels(SNP$FMO3geno)=="4"] <- "HA"
levels(SNP$FMO3geno)[levels(SNP$FMO3geno)=="5"] <- "HH"
levels(SNP$FMO3geno)[levels(SNP$FMO3geno)=="6"] <- "HB"
levels(SNP$FMO3geno)[levels(SNP$FMO3geno)=="7"] <- "BA"
levels(SNP$FMO3geno)[levels(SNP$FMO3geno)=="8"] <- "BH"
levels(SNP$FMO3geno)[levels(SNP$FMO3geno)=="9"] <- "BB"

# Make a new variable that counts the number of risk alleles
# variables are coded as 1, 2, 3
# 1=A homo wildtype; 2=H heterozygous; 3=B homo variant
# in terms of how many variant alleles each geno has
# A=0 risk alleles, H=1 risk allele, B=2 risk alleles; use this to calculate new variable
#rs2266782 A H B
#rs2266780 A H B
# AA AH AB HA HH HB BA BH BB
# 0  1  3  1  2  3  2  3  4
# now group to help you code the ifelse statement
SNP$FMO3_nRiskAlleles <- ifelse(SNP$rs2 == 1 & SNP$rs0 == 1, 0,
                       ifelse(SNP$rs2 == 1 & SNP$rs0 == 2, 1,
                              ifelse(SNP$rs2 == 1 & SNP$rs0 == 3, 3,
                                     ifelse(SNP$rs2 == 2 & SNP$rs0 == 1, 1,
                                            ifelse(SNP$rs2 == 2 & SNP$rs0 == 2, 2,
                                                   ifelse(SNP$rs2 == 2 & SNP$rs0 == 3, 3,
                                                          ifelse(SNP$rs2 == 3 & SNP$rs0 == 1, 2,
                                                                 ifelse(SNP$rs2 == 3 & SNP$rs0 == 2, 3,
                                                                        ifelse(SNP$rs2 == 3 & SNP$rs0 == 3, 4, 10)
                                                                        )
                                                                 )
                                                          )
                                                   )
                                            )
                                     )
                              )
                       )
# Factor
SNP$FMO3_nRiskAlleles <- as.factor(SNP$FMO3_nRiskAlleles)

# And a variable for each individual SNP... WT or at least 1 variant
# Motivation here is that the variant group numbers get small when we look per sex or per sex/age bin
#
# WT + Variant-----------------------------
# Make variable codes for wild type (WT) or variant (V)
# Combine 2 and 3
SNP$rs0_WT_V <- ifelse(SNP$rs0 == 1, 1, 2) #1=WT, 2=at least 1 variant allele
SNP$rs2_WT_V <- ifelse(SNP$rs2 == 1, 1, 2)
# Factor
#rs0
SNP$rs0_WT_V <- factor(SNP$rs0_WT_V, levels = c(1,2))
str(SNP$rs0_WT_V)
#rs2
SNP$rs2_WT_V <- factor(SNP$rs2_WT_V, levels = c(1,2))
str(SNP$rs2_WT_V)
```

Add ethnicity and merge all data

```{r merge dfwide and snp data}
# Merge dfwide and SNP data
dfwide <- merge(dfwide, SNP, by = 0, all.x = TRUE, all.y = FALSE)
rownames(dfwide) <- dfwide[,1]
dfwide <- subset(dfwide, select=-c(Row.names))

# Merge
#dfwide <- merge(dfwide, eth, by = 0)
#rownames(dfwide) <- dfwide[,1]
#dfwide <- subset(dfwide, select=-c(Row.names))
# 10/13/22 Added already in master dfwide 

# Ethnicity
# Reorder the levels so that it matches the order from the demographic survey
str(dfwide$ethnicity)
table(dfwide$ethnicity)
dfwide$ethnicity <- ordered(dfwide$ethnicity, levels = c("Caucasian", "Hispanic", "African.American", "Asian", "Multi", "Other"))
# This also turns ethnicity into a factor (not a character)

# Label sex variable
str(dfwide$sex)
dfwide$sex <- factor(dfwide$sex, levels = c("Male", "Female"), labels = c("Male", "Female") )
# Male=1 here
```

Are the SNPs in HWE?

```{r HWE}
# SNP rs2266782
table(dfwide$rs2)
#rs2 <- c(AA=140,AB=164,BB=45)
rs2 <- c(AA=30,AB=53,BB=14)
hwe.rs2 <- HWChisq(rs2)
hwe.rs2 
# The p statistic, P=0.289 tells us the observed distribution of the SNPs is not different that the expected distribution. The frequency of the A allele in this cohort ($p) is 41.75%

# SNP rs2266780
table(dfwide$rs0)
#rs0 <- c(AA=237,AB=101,BB=11)
rs0 <- c(AA=62,AB=31,BB=4)
hwe.rs0 <- HWChisq(rs0)
hwe.rs0
# The p statistic, P=0.82 tells us the observed distribution of the SNPs is not different that the expected distribution. The frequency of the A allele in this cohort ($p) is 20%
```

How often are the SNPs in LD? In other words, how often is do people inherit both variants? 
```{r genotype table}
# Create table showing how often the SNPs are inherited together
# Recall allele order is rs2, rs0; A=WT,H=het,B=variant
table(dfwide$FMO3geno)
table(dfwide$rs0, dfwide$rs2)

# Table of snps by peakTMAO group
table(dfwide$rs2, dfwide$peakTMAO)
table(dfwide$rs0, dfwide$peakTMAO)
```


Table: Make generic sex, age, BMI table showing who the participants are in the study
```{r Descriptive Table}
colnames(dfwide)
# Generic Desriptive Table
factorVars <- c("age_bin", "sex_bin") # 1=male, 2=female
vars <- c("age", "bmi_final", "ethnicity", "cystatinc_bd1", "TMAO.0","glucose_bd1_0m", "insulin_bd1_0m", "triglycerides_bd1_0m", "HDL_bd1_0m", "LDL_bd1_0m", "chol_bd1_0m")
table.descriptive <- CreateTableOne(vars = vars,
                                    strata = "sex_bin",
                                    data = dfwide,
                                    factorVars = factorVars)
table.descriptive

```
```{r Save descriptive table}
table.descriptive_save <- print(table.descriptive, quote = FALSE, noSpaces = TRUE, printToggle = FALSE)
write.csv(table.descriptive_save, file = "../../tables/descriptive/FL100-MealResponse_TMAO_participant_characteristics.csv")
```


Table: Make table of AUC_TMAO by genotype. Use in manuscript.
```{r}
colnames(dfwide)
str(dfwide$sex)
str(dfwide$rs0)
# TMAO table by genotype
# rs2
factorVars <- c("rs0", "rs2", "FMO3geno")
vars <- c("age", "bmi_final","sex", "peakTMAO", "ethnicity", "cystatinc_bd1", "TMAO.0","TMAO.30", "TMAO.180", "TMAO.360", "AUC_TMAO_uM_hour")
table.geno.rs2 <- CreateTableOne(vars = vars,
                                    strata = "rs2",
                                    data = dfwide,
                                    factorVars = factorVars)
table.geno.rs2

# rs0
table.geno.rs0 <- CreateTableOne(vars = vars,
                                    strata = "rs0",
                                    data = dfwide,
                                    factorVars = factorVars)
table.geno.rs0

# combined
vars2 <- c("sex", "TMAO.0","TMAO.30", "TMAO.180", "TMAO.360", "AUC_TMAO_uM_hour")
table.geno.FMO3geno <- CreateTableOne(vars = vars2,
                                    strata = "FMO3geno",
                                    data = dfwide,
                                    factorVars = factorVars)

table.geno.FMO3geno

# Add AUC values to table in manuscript
# Manually calculate averages
(42.01 + 28.17 + 48.54) / 3
(31.98 + 60.24) / 2
(28.8 + 35.86) / 2
(28.17 + 31.98 + 35.86) / 3
(48.54 + 60.24 + 28.8 ) / 3
```


```{r Save genotype table}
# rs2
table.geno.rs2_save <- print(table.geno.rs2, quote = FALSE, noSpaces = TRUE, printToggle = FALSE)
write.csv(table.geno.rs2_save, file = "../../tables/descriptive/FL100-MealResponse-TMAO_genotype_rs2266782_characteristics.csv")

# rs0
table.geno.rs0_save <- print(table.geno.rs0, quote = FALSE, noSpaces = TRUE, printToggle = FALSE)
write.csv(table.geno.rs0_save, file = "../../tables/descriptive/FL100-MealResponse-TMAO_genotype_rs2266780_characteristics.csv")

# FMO3geno
table.geno.FMO3geno_save <- print(table.geno.FMO3geno, quote = FALSE, noSpaces = TRUE, printToggle = FALSE)
write.csv(table.geno.FMO3geno_save, file = "../../tables/descriptive/FL100-MealResponse-TMAO_genotype_FMO3geno_characteristics.csv")
```

Plot genotype by AUC
rs2266782
```{r plot rs2266782 by AUC TMAO}
p_sex <- ggplot(data = dfwide, aes(x = rs2, y = log(AUC_TMAO_uM_hour))) + 
  geom_boxplot(aes(fill=rs2)) + #aes(fill=rs0_WT_V)) +
  geom_point(aes(fill=rs2), color = "black", alpha = 0.4, position = position_jitterdodge()) +
  facet_grid(. ~ sex) +
  scale_fill_brewer(palette=c("RdYlBu"),
                    name = "rs2266782 \nGenotype",
                    labels =c("GG","GA","AA")) +
  theme_bw() + 
  theme(axis.title=element_text(size=14),
        axis.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.text=element_text(size=12)) + 
  xlab("") +
  ylab(bquote("Ln AUC-TMAO (" ~mu~ "M/hr )")) +
  scale_x_discrete(breaks=c("1", "2", "3"),
        labels=c("GG","GA","AA")) + 
  stat_compare_means(method = "anova", paired=F, label.x = .8)
 # stat_compare_means(comparisons = list(c("1","2"), c("1","3"), c("2","3")), 
 # label = "p.format", method = "wilcox.test", paired=F)
  
p_sex

p <- ggplot(data = dfwide, aes(x = rs2, y = log(AUC_TMAO_uM_hour))) + 
  geom_boxplot(aes(fill=rs2)) + #aes(fill=rs0_WT_V)) +
  geom_point(aes(fill=rs2), color = "black", alpha = 0.4, position = position_jitterdodge()) +
  #facet_grid(. ~ sex) +
  scale_fill_brewer(palette=c("RdYlBu")) +
  #ggtitle("TMAO by FMO3 SNP rs2266780 and Menopausal Status")+
  theme_bw() + 
  theme(strip.text.x = element_text(size = 12),
        text = element_text(size=14)) +
  xlab("") +
  ylab(bquote("Ln AUC TMAO (" ~mu~ "M x hr^-1)")) +
  scale_x_discrete(breaks=c("1", "2", "3"),
        labels=c("GG","GA","AA")) + 
  labs(fill = "rs2266782 \nGenotype") +
  stat_compare_means(comparisons = list(c("1","2"), c("1","3"), c("2","3")), 
  label = "p.format", method = "wilcox.test", paired=F)
  
p
```

```{r save rs2266782 plot}
# Save
ggsave("../../plots/TMAO_AUC_rs2266782_sex_Barbplot_v2.jpg", p_sex,
       width = 8, 
       height = 6, 
       dpi = 300)

ggsave("../../plots/TMAO_AUC_rs2266782_Barbplot.jpg", p,
       width = 8, 
       height = 6, 
       dpi = 300)

```
rs2266780
```{r plot rs2266780 by AUC TMAO}
p_sex <- ggplot(data = dfwide, aes(x = rs0, y = log(AUC_TMAO_uM_hour))) + 
  geom_boxplot(aes(fill=rs0)) + #aes(fill=rs0_WT_V)) +
  geom_point(aes(fill=rs0), color = "black", alpha = 0.4, position = position_jitterdodge()) +
  facet_grid(. ~ sex) +
  scale_fill_brewer(palette=c("RdYlBu")) +
  scale_fill_brewer(palette=c("RdYlBu"),
                    name = "rs2266780 \nGenotype",
                    labels=c("AA","AG","GG")) +
  #ggtitle("TMAO by FMO3 SNP rs2266780 and Menopausal Status")+
  theme_bw() + 
  theme(axis.title=element_text(size=14),
        axis.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.text=element_text(size=12)) + 
  xlab("") +
  ylab(bquote("Ln AUC TMAO (" ~mu~ "M/hr )")) +
  scale_x_discrete(breaks=c("1", "2", "3"),
        labels=c("AA","AG","GG")) + 
  labs(fill = "rs2266780 \nGenotype") +
  stat_compare_means(method = "anova", paired=F, label.x = .8)
  #stat_compare_means(comparisons = list(c("1","2"), c("1","3"), c("2","3")), 
  #label = "p.format", method = "wilcox.test", paired=F)
  
p_sex

p <- ggplot(data = dfwide, aes(x = rs0, y = log(AUC_TMAO_uM_hour))) + 
  geom_boxplot(aes(fill=rs0)) + #aes(fill=rs0_WT_V)) +
  geom_point(aes(fill=rs0), color = "black", alpha = 0.4, position = position_jitterdodge()) +
  #facet_grid(. ~ sex) +
  scale_fill_brewer(palette=c("RdYlBu")) +
  #ggtitle("TMAO by FMO3 SNP rs2266780 and Menopausal Status")+
  theme_bw() + 
  theme(strip.text.x = element_text(size = 12),
        text = element_text(size=14)) +
  xlab("") +
  ylab(bquote("Ln AUC TMAO (" ~mu~ "M x hr^-1)")) +
  scale_x_discrete(breaks=c("1", "2", "3"),
        labels=c("AA","AG","GG")) + 
  labs(fill = "rs2266780 \nGenotype") +
  stat_compare_means(comparisons = list(c("1","2"), c("1","3"), c("2","3")), 
  label = "p.format", method = "wilcox.test", paired=F)
  
p
```

```{r save rs2266780 plot}
# Save
ggsave("../../plots/TMAO_AUC_rs2266780_sex_Barbplot_v2.jpg", p_sex,
       width = 8, 
       height = 6, 
       dpi = 300)

ggsave("../../plots/TMAO_AUC_rs2266780_Barbplot.jpg", p,
       width = 8, 
       height = 6, 
       dpi = 300)
```
Combined geno plots
1st letter corresponds to rs2266782 G>A
A=GG
H=GA
B=AA
2nd letter corresponds to rs2266780 A>G
A=AA
H=AG
B=GG
```{r}
p_allele <- ggplot(data = dfwide, aes(x = FMO3geno, y = log(AUC_TMAO_uM_hour))) + 
  geom_boxplot(aes(fill=FMO3geno)) + #aes(fill=rs0_WT_V)) +
  geom_point(aes(fill=FMO3geno), color = "black", alpha = 0.4, position = position_jitterdodge()) +
  facet_grid(. ~ sex) +
  theme_bw() + 
  theme(axis.title=element_text(size=14),
        axis.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.text=element_text(size=12)) + 
  xlab("") +
  ylab(bquote("Ln AUC TMAO (" ~mu~ "M/hr )")) +
  scale_x_discrete(breaks=c("AA", "HA", "HH", "HB", "BA", "BH", "BB"),
        labels=c("GG:AA", "GA:AA", "GA:AG", "GA:GG", "AA:AA", "AA:AG","AA:GG")) + 
  #labs(fill = "FMO3 \nGenotype", labels=c("GG:AA", "GA:AA", "GA:AH", "GA:GG", "AA:AA", "AA:AG","AA:GG")) +
  scale_fill_brewer(palette=c("RdYlBu"),
                    name = "rs2266782 (G>A): \nrs2266780 (A>G) \nGenotype",
                    labels=c("GG:AA", "GA:AA", "GA:AH", "GA:GG", "AA:AA", "AA:AG","AA:GG")) +
  stat_compare_means(method = "anova", paired=F, label.x = .8)
  #stat_compare_means(comparisons = list(c("AA","HA"), c("AA","HH"), c("AA","HB"), c("AA", "BA"), c("AA", "BH"), c("AA", "BB"),
  #                                      c("HA","HH"), c("HA","HH"), c("HA","HB"), c("HA","BA"), c("HA", "BH"), c("HA", "BB"),
  #                                      c("HH", "HB"), c("HH","BA"), c("HH","BH"), c("HH","BB"),
  #                                      c("HB", "BA"), c("HB","BH"), c("HB", "BB"),
  #                                      c("BA", "BH"), c("BA","BB"),
  #                                      c("BH", "BB")), 
  #label = "p.format", method = "wilcox.test", paired=F)
  
p_allele

ggsave("../../plots/TMAO_AUC_RMO3geno_Barbplot_v3.jpg", p_allele,
       width = 12, 
       height = 6, 
       dpi = 300)
```



Ask the basic question, does AUC_TMAO differ genotype?
```{r AUC-TMAO by genotype in all}
colnames(dfwide)
aov_model <- aov(log(AUC_TMAO_uM_hour) ~ FMO3geno + cystatinc_bd1, dfwide)
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs0, dfwide)
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs2, dfwide)
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ FMO3_nRiskAlleles, dfwide)
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs0_WT_V, dfwide)
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs2_WT_V, dfwide)
summary(aov_model)
```
Check in males 
```{r AUC-TMAO by genotype in males}
colnames(dfwide)
aov_model <- aov(log(AUC_TMAO_uM_hour) ~ FMO3geno + cystatinc_bd1, dfwide[dfwide$sex=="Male",])
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs0, dfwide[dfwide$sex=="Male",])
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs2, dfwide[dfwide$sex=="Male",])
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ FMO3_nRiskAlleles, dfwide[dfwide$sex=="Male",])
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs0_WT_V, dfwide[dfwide$sex=="Male",])
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs2_WT_V, dfwide[dfwide$sex=="Male",])
summary(aov_model)
```

Check in females 
```{r AUC-TMAO by genotype in females}
colnames(dfwide)
aov_model <- aov(log(AUC_TMAO_uM_hour) ~ FMO3geno + cystatinc_bd1, dfwide[dfwide$sex=="Female",]) # .03
summary(aov_model)
aov_model <- aov(log(AUC_TMAO_uM_hour) ~ FMO3geno , dfwide[dfwide$sex=="Female",])
summary(aov_model)
TukeyHSD(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs0, dfwide[dfwide$sex=="Female",])
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs2, dfwide[dfwide$sex=="Female",]) # .08
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ FMO3_nRiskAlleles, dfwide[dfwide$sex=="Female",]) # sig
summary(aov_model)
TukeyHSD(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs0_WT_V, dfwide[dfwide$sex=="Female",])
summary(aov_model)

aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs2_WT_V, dfwide[dfwide$sex=="Female",])
summary(aov_model)

# Test dominant model in females where one WT allele can rescue phenotype
table(dfwide$rs2)
dfwide$rs2_dom <- ifelse(dfwide$rs2 == 1 | dfwide$rs2 == 2, "wt", "affected")
aov_model <- aov(log(AUC_TMAO_uM_hour) ~ rs2_dom, dfwide[dfwide$sex=="Female",])
summary(aov_model)
```
Plot sig diff in FMO3geno
```{r FMO3 geno female BA-HA plots}
# Subset to individuals with BA and HA genos
dfwide_FMO3sub <- dfwide[dfwide$FMO3geno == "BA" | dfwide$FMO3geno == "HA",]
dfwide_FMO3sub.f <- dfwide_FMO3sub[dfwide_FMO3sub$sex == "Female",]

FMO3sub_plot <- ggplot(dfwide_FMO3sub.f, 
                       aes(x = FMO3geno,
                       y = AUC_TMAO_uM_hour,
                       fill = FMO3geno)) +
    geom_boxplot(notch = F) +
    scale_fill_brewer(palette = "RdYlBu",
                      name = "FMO3 Genotype \nrs2266782 (G>A): \nrs2266780 (A>G)",
                      breaks = c("BA", "HA"),
                      labels = c("AA:AA","GA:AA")) +
    xlab(paste0("FMO3 Genotype")) +
    ylab("TMAO uM/hr") +
    geom_jitter() +
    scale_x_discrete(breaks=c("BA", "HA"),
                     labels=c("AA:AA","GA:AA")) +
    theme_bw() +
    theme(axis.title=element_text(size=14),
        axis.text=element_text(size=12),
        legend.title=element_text(size=14),
        legend.text=element_text(size=12)) +
    stat_compare_means(label = "p.format", method = "anova", label.x = 1) 

FMO3sub_plot

ggsave("../../plots/TMAOxTime_Boxplot_FMO3geno_females_BAvHA.jpg",
       width = 5, 
       height = 4, 
       dpi = 300)
  
```


Is genotype distribution different by peakTMAO group?

```{r geno distribution by peak TMAO group}
colnames(dfwide)
str(dfwide$rs2)
str(dfwide$peakTMAO)
# rs2
chisqu_model_rs0 <- chisq.test(dfwide$rs2, dfwide$peakTMAO)
chisqu_model_rs0
chisqu_model_rs0$observed
chisqu_model_rs0$expected

# rs0
chisqu_model_rs0 <- chisq.test(dfwide$rs0, dfwide$peakTMAO)
chisqu_model_rs0
```

Show the mean + SD of male FMO3 all
```{r summary rs2 rs0 }
# Males
# median
# rs2
median(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 1], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 2], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 3], na.rm = TRUE)

# rs0
median(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 1], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 2], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 3], na.rm = TRUE)

# summary (mean, mdn, iqr...)
# rs2
summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 1], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 2], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 3], na.rm = TRUE)

# rs0
summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 1], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 2], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 3], na.rm = TRUE)
```

Show the mean + SD of male FMO3 males

```{r summary rs2 rs0 males}
# Males
# median
# rs2
median(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 1 & dfwide$sex == "Male"], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 2 & dfwide$sex == "Male"], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 3 & dfwide$sex == "Male"], na.rm = TRUE)

# rs0
median(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 1 & dfwide$sex == "Male"], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 2 & dfwide$sex == "Male"], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 3 & dfwide$sex == "Male"], na.rm = TRUE)

# summary (mean, mdn, iqr...)
# rs2
summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 1 & dfwide$sex == "Male"], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 2 & dfwide$sex == "Male"], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 3 & dfwide$sex == "Male"], na.rm = TRUE)

# rs0
summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 1 & dfwide$sex == "Male"], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 2 & dfwide$sex == "Male"], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 3 & dfwide$sex == "Male"], na.rm = TRUE)
```

```{r summary rs2 rs0 females}
# Females
# rs2
median(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 1 & dfwide$sex == "Female"], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 2 & dfwide$sex == "Female"], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 3 & dfwide$sex == "Female"], na.rm = TRUE)

# rs0
median(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 1 & dfwide$sex == "Female"], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 2 & dfwide$sex == "Female"], na.rm = TRUE)

median(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 3 & dfwide$sex == "Female"], na.rm = TRUE)

# summary (mean, mdn, iqr...)
# rs2
summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 1 & dfwide$sex == "Female"], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 2 & dfwide$sex == "Female"], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs2 == 3 & dfwide$sex == "Female"], na.rm = TRUE)

# rs0
summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 1 & dfwide$sex == "Female"], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 2 & dfwide$sex == "Female"], na.rm = TRUE)

summary(dfwide$AUC_TMAO_uM_hour[dfwide$rs0 == 3 & dfwide$sex == "Female"], na.rm = TRUE)
```





Are the peakTMAO groups different by genotype?

```{r chisquare TMAOgroup and FMO3}
chisq.test(dfwide$peakTMAO, dfwide$FMO3geno)
chisq.test(dfwide$peakTMAO, dfwide$rs2)
chisq.test(dfwide$peakTMAO, dfwide$rs0)
chisq.test(dfwide$peakTMAO, dfwide$FMO3_nRiskAlleles)
chisq.test(dfwide$peakTMAO, dfwide$rs0_WT_V)
chisq.test(dfwide$peakTMAO, dfwide$rs2_WT_V)

```

Plot AUC-TMAO by genotype
```{r}
# rs0 plot
rs0_plot <- ggplot(dfwide, aes(x = rs0, #FMO3geno,
                                     y = AUC_TMAO_uM_hour,
                                     fill = rs0)) +
    stat_boxplot(geom='errorbar', linetype=1, width=0.2)+
    geom_boxplot(notch = F) +
    #facet_wrap(~sex) +
    #scale_fill_manual(legend_title, values=c("#2D708EFF", "#B8DE29FF", "#DCE319FF", "#E0E0E0")) +
    xlab(paste0("Glu308Gly Genotype")) +
    ylab("TMAO uM/hr") +
    geom_jitter() +
    theme_bw() +
    theme(plot.title = element_text(hjust = 0.5),
          text = element_text(family = "Arial",
                              colour = "black")) +
    theme(panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          plot.title = element_text(size = 10),
          axis.title.y=element_text(size=11,
                                    colour = "black"),
          axis.text.y=element_text(size=10,
                                   colour = "black")) +
    stat_compare_means(label = "p.format", method = "anova" ,label.y = 6) +
      scale_x_discrete(breaks=c("1","2","3"),
                     labels=c("AA","AG", "GG"))
rs0_plot

# rs2 plot, Glu158Lys
rs2_plot <- ggplot(dfwide, aes(x = rs2, #FMO3geno,
                                     y = AUC_TMAO_uM_hour,
                                     fill = rs2)) +
    stat_boxplot(geom='errorbar', linetype=1, width=0.2)+
    geom_boxplot(notch = F) +
    #facet_wrap(~sex) +
    #scale_fill_manual(values=c("#2D708EFF", "#B8DE29FF", "#DCE319FF", "#E0E0E0")) +
    scale_fill_brewer(palette = "RdYlBu") +
    xlab(paste0("Glu158Lys Genotype")) +
    ylab("TMAO uM/hr") +
    geom_jitter() +
    theme_bw() +
    theme(plot.title = element_text(hjust = 0.5),
          text = element_text(family = "Arial",
                              colour = "black")) +
    theme(panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          plot.title = element_text(size = 10),
          axis.title.y=element_text(size=11,
                                    colour = "black"),
          axis.text.y=element_text(size=10,
                                   colour = "black")) +
    stat_compare_means(label = "p.format", method = "anova" ,label.x = 2) +
    scale_x_discrete(breaks=c("1","2","3"),
                     labels=c("AA","AG", "GG"))
rs2_plot
```


